src/tracer.js

```diff
@@ -253,6 +265,28 @@ export default class Tracer {
     }
 
     /**
+     * Reapplies sampling and updates parent's sampling flag
+     *
+     * @param parent the parent span context
+     * @param ctx the span context
+     * @param operationName the operation name for the child
+     * @param tags tags to be applied by the sampler
+     */
+    handleUpsampling(parent: SpanContext, ctx: SpanContext, operationName: string, tags: any) {
+        if (!this._upsampling.enabled) {
+            return
+        }
+
+        let samplingDecision = this._sampler.isSampled(operationName, tags);
+
+        parent.samplingDecision = samplingDecision;
+        parent.finalizeSampling();
```

Comment:
why are we touching parent here? If this span is first in process, then parent is irrelevant, but not first in process, then we will inherit parent's decision without upsampling

Reply:
Agree, updated.